src/Microsoft.Azure.ServiceBus/Amqp/AmqpExceptionHelper.cs

```diff
@@ -192,7 +194,7 @@ public static Exception GetClientException(Exception exception, string reference
                     return new ServiceBusException(true, message, exception);
 
                 case TimeoutException _ when referenceId != null:
```

Comment:
This case is true only when referenceId!=null.
Cases when referenceId is null, we still end up throwing `TimeoutException` instead of `ServiceBusTimeoutException`. Please take care of that scenario.